src/reason_pprint_ast.ml

```diff
@@ -4017,11 +4094,21 @@ class printer  ()= object(self:'self)
   method expression x =
     let (arityAttrs, docAtrs, stdAttrs) = partitionAttributes x.pexp_attributes in
     if stdAttrs <> []  && (not (self#expr_can_render_own_attributes x)) then
+      let rec findJSXAttribute attrs =
+        match attrs with
+          | ({txt = "JSX"; _}, _) :: tail -> true
+          | _ :: tail -> findJSXAttribute tail
+          | [] -> false
+      in
+      let hasJSXAttribute = findJSXAttribute x.pexp_attributes in
       let withoutVisibleAttrs = {x with pexp_attributes=arityAttrs} in
         let itm =
-          formatAttributed
-            (self#simple_expression withoutVisibleAttrs)
-            (self#attributes stdAttrs)
+          if hasJSXAttribute then
```

Comment:
I think this is supposed to say `if theOnlyStdAttrsWereJsxAttrs then` (if it had `stdAttrs` but all of them were `JSX` attrs).
Otherwise, as it is, I think sometimes attributes would get dropped from JSX expressions.